temp_name_lib/src/star_physics/intensity.rs

```diff
@@ -0,0 +1,71 @@
+use nalgebra as na;
+
+use crate::utils::MathErrors;
+
+enum GridPointStatus{
+    NotAvailable,
+    NotLoaded,
+    Loaded,
+}
+
+enum IntensityType{
+    Total,
+    ContinuumOnly
+}
+
+pub struct Intensity {
+    temp_eff:Vec<f64>,
+    log_g:Vec<f64>,
+    lambda:Vec<f64>,
+    file_name:String,//[Ricardo]: I suppose you have different intensity files.
+    
+    temp_eff_index:usize,
+    log_g_index:usize,
+    lambda_index:usize,
+
+    lambda_lower:f64,
+    lambda_upper:f64,
+    limb_darkening_coef:na::Vector4<f64>,
+    grid_status:GridPointStatus,
+    kind:IntensityType,
+}
+
+
+impl Intensity{
+
+    pub fn read_file(){}
```

Comment:
The Rustacean way to name this is `from_file()` so that you can write 
`intensities = Intensity::from_file(path)`

Note, however, that what you want is what is called an `InterpolatingFunction`: an object that behaves like a function and does not need to make the user know that it has an underlying grid.